File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* removeNthFromEnd(struct ListNode* head, int n){
  struct ListNode* newHead = (struct ListNode*)malloc(sizeof(struct ListNode));
  newHead->val = -1;
  newHead->next = head;
  int gap = 0;
  struct ListNode *first = newHead, *second = newHead;
  while(second){
    if(gap != n+1){
      gap++;
      second = second->next;
      continue;
    }
    first = first->next;
    second = second->next;
  }
  first->next = first->next->next;
  return newHead->next;
}

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.ts
function removeNthFromEnd(head: ListNode | null, n: number): ListNode | null {
  var newHead : ListNode = new ListNode(-1, head);
  var gap: number = 0;
  var first: ListNode = newHead;
  var second: ListNode = newHead;
  while(second){
    if(gap != n+1){
      gap++;
      second = second.next;
      continue;
    }
    first = first.next;
    second = second.next;
  }
  first.next = first.next.next;
  return newHead.next;
};